### ISSUE TITLE
Battery prompt in Termux: permission denied

### ISSUE THREAD

**Issue by toloveru** [bhilburn/powerlevel9k#820]
I'm using Powerlevel9k as my theme in Zsh, with the battery indicator enabled. As I'm trying to make my [dotfiles](https://github.com/toloveru/dotfiles/tree/develop) compatible with all my devices, I've stumbled upon the following error that's displayed every time I run zsh on my Android 7.0 tablet.
`cat: can't open '/sys/class/power_supply/battery/capacity': Permission denied`
`cat: can't open '/sys/class/power_supply/battery/status': Permission denied`
![screenshot_20180503-015352](https://user-images.githubusercontent.com/19938179/39556085-07302222-4e7e-11e8-8aa1-35e2494e075b.png)

So it can't read from the sysfs due to permission issues. When I'm executing `su -c cat /sys/class/power_supply/battery/capacity` instead, it does display the capacity (well, remaining capacity) properly. Same goes for the other one. I've been able to narrow this down to lines 441 and 442 in powerlevel9k.zsh-theme, but changing `local capacity=$(cat $bat/capacity)` to `local capacity=$(su -c cat $bat/capacity)` (same goes for the second line) threw up different errors. This made it appear that this modification either caused other functions or variables in that file, or other sourced files to break. At this point I stopped investigating because it appeared that my Vim inside Termux threw in unwanted characters into these files during edits (which weren't saved - I exited with :qa!, but irritating nonetheless).

Basically I'd like powerlevel9k to be able to notice when Termux or Android is being used, and make the battery function either run as superuser, or leverage Termux:API's `termux-battery-status` (second one would be preferred) when that's the case.

Thanks in advance for looking into this!

**Comment by dritter** [comment]
Hi @toloveru !
Thanks for the report and the insights. I am opposed to add `su` to the commands.. The capacity should be a information available to all users. Whom does your `capacity` file belong? And which access is allowed on this file?
Does it work to access other files in `/sys/class/`?

I am not a fan of adding another method to get the battery status. This segment is already complicated enough.. Hoping for somebody with a good alternative..

Honestly Andoid is based on SELinux, which secures most accesses to system files. So it might happen on other segments as well.

**Comment by toloveru** [comment]
Sorry for the late reply - the issue was indeed caused by SELinux. When setting the mode to permissive (`su -c setenforce 0`) it became possible to read out the file. So this issue is solved. Thanks a lot for pointing me in the right direction!
